README.md

```diff
@@ -1,116 +1,170 @@
+- **Configuration Classes**: Both models utilize specific configuration classes (`RetNetConfig` for RetNetModelHF and `DecoderConfig` for TransformerModelHF) to define model parameters, ensuring flexibility and ease of customization.
+
+**Usage:**
+
+To use these models within your Hugging Face-based projects, follow these steps:
+
+1. **Initialization**: Instantiate the model with the desired configuration, which can be a predefined object, a path to a configuration file, or left as default for automatic configuration.
+
+   ```python
+   from <YOUR_MODULE> import RetNetModelHF, TransformerModelHF
+
+   retnet_model = RetNetModelHF(config="path/to/retnet/config")
+   transformer_model = TransformerModelHF(config="path/to/transformer/config")
+   ```
+
+2. **Forward Pass**: Call the model with input data tensors to receive output predictions.
+
+   ```python
+   input_ids = ...  # Your input tensor here
+   retnet_output = retnet_model(input_ids)
+   transformer_output = transformer_model(input_ids)
+   ```
+
+3. **Parameter Access**: Retrieve model hyperparameters for inspection or further processing.
+
+   ```python
+   retnet_params = retnet_model.get_params()
+   transformer_params = transformer_model.get_params()
+   ```
+
+## Benchmarking and Results
+
+We use EleutherAI's open-source language model evaluation harness to empirically evaluate our models across a suite of different NLP tasks. Run the evaluation suite as follows:
+First, edit the 'tasks' parameter in the YAML file. Specify all tasks you would like to run, e.g.,
+```
+tasks:
+  - "hellaswag"
+  - "winogrande"
 ```
+Alternatively, you can use `tasks: '*'` to run all benchmarks in the suite. Then navigate to the `slurm/run_eval.sh`, copy the script, and substitute your yaml file for the placeholder. Finally, execute:
 
-In `config.yaml`, you can have:
+```
+# Activate environment, if using one
+mamba activate <YOUR_ENV_HERE>
+
+cd /301r_retnet/slurm/
+
+# Give your file a descriptive name, (e.g., 'retnet_40540_run_eval.sh')
+cp run_eval.sh user_slurm/<NAME_OF_NEW_FILE>.sh
 
-```yaml
-learning_rate: 0.001
-batch_size: 64
-num_layers: 4
-embed_dim: 256
+bash <NAME_OF_NEW_FILE>/.sh
 ```
+Results will be sent to a CSV.
+
+## Acknowledgments
+
+We extend our heartfelt gratitude to the following individuals and institutions for their invaluable contributions to this project:
+
+**Nancy Fulda**: Our esteemed instructor, whose guidance and insights have significantly shaped the direction and execution of this research.
+
+**BYU Office of Research Computing**: For providing the computational resources and support that were instrumental in conducting our experiments and analyses.
+
+**Authors of the Original RetNet Paper**: For their pioneering work in the field of deep learning, which inspired our research and provided a solid base for our explorations into RetNet and Transformers.
```

Comment:
I would not say "pioneering work in the field of deep learning" --> that feels too generous, in that it is false. RetNet is not truly pioneering work in the field of deep learning.
Perhaps "For their work in novel encoder architectures"?